internal/dyngo/operation.go

```diff
@@ -0,0 +1,91 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package dyngo
+
+type (
+	Option interface {
+		apply(s *Operation)
+	}
+
+	optionFunc func(s *Operation)
+)
+
+func (f optionFunc) apply(s *Operation) {
+	f(s)
+}
+
+func WithParent(parent *Operation) Option {
+	return optionFunc(func(op *Operation) {
+		op.parent = parent
+	})
+}
```

Comment:
Why make `Option` an interface rather than just a function type, like we do [everywhere else](https://github.com/DataDog/dd-trace-go/blob/f3f95f78ef7c7904c33856da78130515fa01266c/ddtrace/tracer/option.go#L112-L113)?

Reply:
This is a way to enforce every option to be defined in this package because the interface method is private so it can only be implemented here in `dyngo`. While having only a `type Option func(*Operation)` can be implemented everywhere.

This is Go pattern is commonly used in the standard library.

Reply:
I tend to agree with Kyle here. The interface approach is also used in the `opentracing-go` repository, but honestly I haven't seen it anywhere else. It is too abstract and complicated. It adds also more boilerplate to introduce a new option. A simple function type should do. This use does not qualify for an interface. My 2c.

Reply:
After spending the past few days making dyngo's listeners statically typed to obtain compile-time errors, I feel this change is contradictory as it would make this possible `dyngo.StartOperation(Args{}, func(op *Operation){ /* ... */ })`, while this slightly more verbose implementation makes this impossible and guarantees dyngo is the only option provider. So I really think this static enforcement is worth it.

Reply:
You can make the config argument private. Then, nobody can write new options, if that's what you want.

Reply:
Now removed as there were no other option than the parent, I made it an argument of the StartOperation function, to emphasis the fact it should because we don't have any way to automatically get the current parent for now.

Note: we should add in our style-guide that this Go design pattern is forbidden for the sake of code readability.